Feat: App start span for Android (#3739)
getsentry/sentry-docs: src/platforms/android/common/performance/instrumentation/automatic-instrumentation.mdx

When you didn't finish the transaction yet, but you start a new Activity, the SDK always automatically finishes the previous Activity transaction. This is due to the fact that only one transaction at a time can be bound to the Scope. To work around that, you may create transactions manually using the custom instrumentation and its instance to start spans instead of the Static API.

### App Start Instrumentation

<Note>

This is a preview API, available in `5.1.0-beta.1` and above. Features available in a preview API are still in progress and may have bugs. We recognize the irony. The API may be renamed, changed, or removed in a future version.

</Note>

The App Start Instrumentation provides insight into how long your application takes to launch. It adds a span from the application launch to the first auto-generated UI transaction.

> To enable this feature enable `enableAutoActivityLifecycleTracing`. The SDK differentiates between a cold and a warm start, and doesn't track hot starts/resumes.

Author: ```suggestion
The SDK differentiates between a cold and a warm start, and doesn't track hot starts/resumes.
```

> * __Cold start__: App launched for the first time, after a reboot or update. The app is not in memory and no process exists.

Author: ```suggestion
* __Cold start__: A cold start refers to an app’s starting from scratch: the system’s process has not, until this start, created the app’s process. Cold starts happen in cases such as your app’s being launched for the first time since the device booted, or since the system killed the app.
```

> * __Warm start__: App launched at least once, is partially in memory, and no process exists.

Reviewer: As you pointed out in one of our calls, I think Android might cache the process, and it is not created when you start the app. Also, the [Android docs](https://developer.android.com/topic/performance/vitals/launch-time#warm) on warm start:

> The user backs out of your app, but then re-launches it. The process may have continued to run, but the app must recreate the activity from scratch via a call to onCreate().

Author: good point, thanks, I borrowed the sentence from the android docs for cold/warm.

> * __Warm start__: App launched at least once, is partially in memory, and no process exists.

Author: ```suggestion
* __Warm start__: A warm start encompasses some subset of the operations that take place during a cold start; at the same time, it represents more overhead than a hot start. For instance: The user backs out of your app, but then re-launches it. The process may have continued to run, but the app must recreate the activity from scratch via a call to `onCreate()`.

```

When you didn't finish the transaction yet, but you start a new Activity, the SDK always automatically finishes the previous Activity transaction. This is due to the fact that only one transaction at a time can be bound to the Scope. To work around that, you may create transactions manually using the custom instrumentation and its instance to start spans instead of the Static API.

### App Start Instrumentation

<Note>

This is a preview API, available in `5.1.0-beta.1` and above. Features available in a preview API are still in progress and may have bugs. We recognize the irony. The API may be renamed, changed, or removed in a future version.

</Note>

The App Start Instrumentation provides insight into how long your application takes to launch. It adds a span from the application launch to the first auto-generated UI transaction.

> The SDK differentiates between a cold and a warm start, and doesn't track hot starts/resumes.

Reviewer 2: ```suggestion
The SDK differentiates between a cold and a warm start, but doesn't track hot starts/resumes.
```

> * __Cold start__: A cold start refers to an app’s starting from scratch: the system’s process has not, until this start, created the app’s process. Cold starts happen in cases such as your app’s being launched for the first time since the device booted, or since the system killed the app.
> * __Warm start__: A warm start encompasses some subset of the operations that take place during a cold start; at the same time, it represents more overhead than a hot start. For instance: The user backs out of your app, but then re-launches it. The process may have continued to run, but the app must recreate the activity from scratch via a call to `onCreate()`.

Reviewer 2: (I'm using the Apple definitions as a base here)
```suggestion
* __Cold start__: The app is launching for the first time, after a reboot or update. The app is not in memory and no process exists.
* __Warm start__: Some subset of the operations that take place during a cold start; representing more overhead than a hot start. For instance: The user backs out of your app, then re-launches it. The process may have continued to run, but the app must recreate the activity from scratch via a call to `onCreate()`.
```

Author: I can't use the iOS spec, this is a specific per platform, they track things differently, hence it requires the description based on the OS

Author: I will keep as it is then
